platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/WrappedPersistenceManager.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.felix.cm.file;
+
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.Enumeration;
+
+import org.apache.felix.cm.PersistenceManager;
+
+/**
+ * Basic wrapper.
+ */
+public class WrappedPersistenceManager implements PersistenceManager {
+
+    private final PersistenceManager persistenceManager;
+
+    public WrappedPersistenceManager(PersistenceManager persistenceManager) {
```

Comment:
✏️ Should we protect against null? Should the constructor be protected and the class abstract to force it to be extended?

Reply:
Making the class abstract makes testing just a bit more obscure since we will need to create an anonymous child class. Also that strikes me as pre-optimization since for encrypting etc this code will change. 

However, +1 for null safety. And the ctor is now package private.

Reply:
Also added a test for the null safety check.

Reply:
no need to create an anonymous class. Checkout the test classes in the migration API code, you can easily create a mock that delegates everything to the real method and then test with that.